### ISSUE TITLE
dotnet run "eats" empty command line arguments

### ISSUE THREAD

**Issue by VasiliyNovikov** [dotnet/cli#8892]
## Steps to reproduce
Have a simple console app written on C#:
```
using System;

namespace ConsoleEmptyParamTest
{
    class Program
    {
        static void Main(string[] args)
        {
            for(var i = 0; i < args.Length; ++i)
                Console.WriteLine($"Arg {i}: \"{args[i]}\"");
        }
    }
}
```
And equivalent written in Python to compare the behavior:
```
import sys

args = sys.argv[1:] # Exclude Program.py itself
for i in xrange(len(args)):
    print('Arg %d: "%s"' % (i, args[i]))
```
Run them from command line:
`dotnet run a "" "b"`
`python Program.py a "" "b"`

## Expected  behavior
I'd expect the following output for both of them:
```
Arg 0: "a"
Arg 1: ""
Arg 2: "b"
```

## Actual behavior
.NET Core code outputs:
```
Arg 0: "a"
Arg 1: "b"
```
Python code outputs:
```
Arg 0: "a"
Arg 1: ""
Arg 2: "b"
```

## Environment data
`dotnet --info` output on Windows:
```
.NET Command Line Tools (2.1.300-preview1-008174)

Product Information:
 Version:            2.1.300-preview1-008174
 Commit SHA-1 hash:  b8df89a54f

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.300-preview1-008174\

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview1-26216-03
  Build    : f2c3216183d20416568a4bbf5bb7d153e826f153
```
`dotnet --info` output on Linux:
```
.NET Command Line Tools (2.1.300-preview1-008174)

Product Information:
 Version:            2.1.300-preview1-008174
 Commit SHA-1 hash:  b8df89a54f

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  16.04
 OS Platform: Linux
 RID:         ubuntu.16.04-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.300-preview1-008174/

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview1-26216-03
  Build    : f2c3216183d20416568a4bbf5bb7d153e826f153
```

## Note
This example crashes at all on latest released version of .NET Core (runtime version 2.0.6) but seems like this crash was fixed for .NET Core 2.1

**Comment by mistachkin** [comment]
I've hit this issue as well.  The culprit appears to be the ArgumentEscaper.EscapeSingleArg() method.  When an empty argument is hit, I *think* it should emit a pair of double quotes.

**Comment by nguerrera** [comment]
Thanks, @mistachkin for the contribution.

I'm wondering if we can move to [ProcessStartInfo.ArgumentList](https://apisof.net/catalog/System.Diagnostics.ProcessStartInfo.ArgumentList) that was added in .NET Core 2.1. This would allow us to have full control over the argv sent on Unix, and to depend on the .NET Framework to do the best effort on Windows.

**Comment by peterhuene** [comment]
Fix was merged and is available in 3.0.100 Preview 4 SDK.
